src/strings_type.h

```diff
@@ -73,4 +74,22 @@ static constexpr StringID SPECSTR_PRESIDENT_NAME = 0x70E7; ///< Special string f
 
 using StringParameterData = std::variant<uint64_t, std::string>;
 
+/** The data required to format and validate a single parameter of a string. */
+struct StringParameter {
+	StringParameterData data; ///< The data of the parameter.
+	char32_t type; ///< The #StringControlCode to interpret this data with when it's the first parameter, otherwise '\0'.
+
+	StringParameter() = default;
+	inline StringParameter(StringParameterData &&data) : data(std::move(data)), type(0) {}
+
+	inline StringParameter(uint64_t data) : data(data), type(0) {}
+
+	inline StringParameter(const char *data) : data(std::string{data}), type(0) {}
+	inline StringParameter(std::string &&data) : data(std::move(data)), type(0) {}
+	inline StringParameter(const std::string &data) : data(data), type(0) {}
+
+	template <ConvertibleThroughBase T>
+	inline StringParameter(const T &data) : data(static_cast<uint64_t>(data.base())), type(0) {}
```

Comment:
```suggestion
	inline StringParameter(const ConvertibleThroughBase auto &data) : data(static_cast<uint64_t>(data.base())), type(0) {}
```
If you like a shorter way of writing it. This is fine as well, and needed when you need to know the type.